--- exact-image-0.9.1/api/perl/Makefile.perl
+++ exact-image-0.9.1/api/perl/Makefile
@@ -5,6 +5,7 @@
 DEPS = $(lib_BINARY) $(bardecode_BINARY) $(codecs_BINARY) $(api_BINARY) $($(X_MODULE)_OUTPUT)/api-perl-wrap.cc
 
 objdir/api/perl/CXXFLAGS := -I api $(PERLINCS)
+objdir/api/perl/LDFLAGS := $(PERLLIBS) -lperl
 
 $($(X_MODULE)_OUTPUT)/api-perl-wrap.cc: $(X_MODULE)/../api.hh $(X_MODULE)/../api-swig.hh
 	$(Q)echo "  SWIG      $(dir $@)"
@@ -16,9 +17,11 @@
 include build/bottom.make
 X_BUILD_IMPLICIT := $(_X_BUILD_IMPLICIT)
 
+PERL_LIBDIR := $(shell perl -MConfig -e 'print $$Config{vendorarch};')
+
 # install
 all:: $($(X_MODULE)_BINARY)
 install:: $($(X_MODULE)_BINARY)
 	$(Q)echo "INSTALL PERL module $^"
-	$(Q)mkdir -p $(DESTDIR)$(libdir)/perl5/site_perl/
-	$(Q)install $^ $(dir $^)/ExactImage.pm $(DESTDIR)$(libdir)/perl5/site_perl/
+	$(Q)mkdir -p $(DESTDIR)$(PERL_LIBDIR)/
+	$(Q)install $^ $(dir $^)/ExactImage.pm $(DESTDIR)$(PERL_LIBDIR)/
