--- exact-image-0.9.1/codecs/dcraw.h.orig
+++ exact-image-0.9.1/codecs/dcraw.h
@@ -4073,7 +4073,7 @@
  */
 void CLASS vng_interpolate()
 {
-  static const signed char *cp, terms[] = {
+  static const signed int *cp, terms[] = {
     -2,-2,+0,-1,0,0x01, -2,-2,+0,+0,1,0x01, -2,-1,-1,+0,0,0x01,
     -2,-1,+0,-1,0,0x02, -2,-1,+0,+0,0,0x03, -2,-1,+0,+1,1,0x01,
     -2,+0,+0,-1,0,0x06, -2,+0,+0,+0,1,0x02, -2,+0,+0,+1,0,0x03,
@@ -6604,7 +6604,7 @@
 {
   static const struct {
     const char *prefix;
-    short black, maximum, trans[12];
+    int black, maximum, trans[12];
   } table[] = {
     { "AgfaPhoto DC-833m", 0, 0,	/* DJC */
 	{ 11438,-3762,-1115,-2409,9914,2497,-1227,2295,5300 } },
@@ -8950,7 +8950,7 @@
   strncpy (th->desc, desc, 512);
   strncpy (th->make, make, 64);
   strncpy (th->model, model, 64);
-  strcpy (th->soft, "dcraw v"DCRAW_VERSION);
+  strcpy (th->soft, "dcraw v"); strcpy (th->soft, DCRAW_VERSION);
   t = localtime (&timestamp);
   sprintf (th->date, "%04d:%02d:%02d %02d:%02d:%02d",
       t->tm_year+1900,t->tm_mon+1,t->tm_mday,t->tm_hour,t->tm_min,t->tm_sec);
--- exact-image-0.9.1/codecs/raw.cc.orig
+++ exact-image-0.9.1/codecs/raw.cc
@@ -71,9 +71,11 @@
   if (!image.getRawData())
     return false;
 
-  return stream->write ((char*)image.getRawData(), image.stride()*image.h)
+  stream->write ((char*)image.getRawData(), image.stride()*image.h)
     /* ==
        (size_t) image.stride()*image.h*/;
+
+  return stream->good();
 }
 
 RAWCodec raw_loader;
