--- exact-image-0.9.1/api/php/Makefile.orig
+++ exact-image-0.9.1/api/php/Makefile
@@ -5,6 +5,7 @@
 DEPS = $(lib_BINARY) $(bardecode_BINARY) $(codecs_BINARY) $(api_BINARY) $($(X_MODULE)_OUTPUT)/api-php-wrap.cc
 
 objdir/api/php/CXXFLAGS := -I api $(PHPINCS) -fno-strict-aliasing
+objdir/api/php/LDFLAGS := $(PHPLIBS)
 
 $($(X_MODULE)_OUTPUT)/api-php-wrap.cc: $(X_MODULE)/../api.hh $(X_MODULE)/../api-swig.hh
 	$(Q)echo "  SWIG      $(dir $@)"
